internal/store/fleet.go

```diff
@@ -187,14 +187,16 @@ func (s *FleetStore) createOrUpdateTx(tx *gorm.DB, ctx context.Context, orgId uu
 			}
 			fleet.Generation = util.Int64ToPtr(1)
 			fleet.Spec.Data.Template.Metadata.Generation = util.Int64ToPtr(1)
-			templateUpdated = true
+			templateUpdated = false // No point in rolling out until selectors are processed
+			selectorUpdated = true
```

Comment:
May be this should be logic in the callback itself?, it's in the end inner knowledge that we first need to process all the devices and update based on selector, then update based on the template content. I'd move it down there.

Reply:
It's just telling the callback what changed, not what to do.  I can leave "templateUpdated=true" to avoid inner knowledge.  It doesn't break anything, just a small optimization.  WDYT?

Reply:
Fixed